pkg/resourcecreator/pod/pod.go

```diff
@@ -235,15 +248,51 @@ func CreateNaisjobObjectMeta(naisjob *nais_io_v1.Naisjob, ast *resource.Ast) met
 	return objectMeta
 }
 
-func lifeCycle(path string) *corev1.Lifecycle {
-	if len(path) > 0 {
+func lifecycle(preStopHookPath string, preStopHook *nais_io_v1.PreStopHook) (*corev1.Lifecycle, error) {
+	if len(preStopHookPath) > 0 && preStopHook != nil {
+		return nil, fmt.Errorf("can only use one of spec.preStopHookPath or spec.preStopHook")
+	}
+
+	if len(preStopHookPath) > 0 {
 		return &corev1.Lifecycle{
 			PreStop: &corev1.Handler{
 				HTTPGet: &corev1.HTTPGetAction{
-					Path: strings.TrimLeft(path, "/"),
+					Path: strings.TrimLeft(preStopHookPath, "/"),
 					Port: intstr.FromString(nais_io_v1alpha1.DefaultPortName),
 				},
 			},
+		}, nil
+	}
+
+	if preStopHook != nil {
+		if preStopHook.Exec != nil && preStopHook.Http != nil {
```

Comment:
Ville vurdert å gjøre håndtering av de forskjellige alternativene til egne funksjoner, sånn at ikke denne blir så veldig lang og "innrykkete".

Reply:
Her er det alt for mye iffing og elsing, og for lite return early. Så i prinsipp enig med Morten, men synes ikke man trenger flere funksjoner.